### PR TITLE
Fix author parsing for `BSZ`

### PR DESCRIPTION
**Pull request by MaxDall**
This PR replaces the current generic implementation of author parsing for `BSZ` as well as replacing the current test case with a more challenging one.

### CODE REVIEW

File: src/fundus/parser/data.py [hunk 1]
```diff
@@ -85,7 +88,7 @@ def get_value_by_key_path(self, key_path: List[str], default: Any = None) -> Opt
             tmp = nxt
         return tmp
 
-    def bf_search(self, key: str, depth: Optional[int] = None, default: Any = None) -> Optional[Any]:
+    def bf_search(self, key: str, depth: Optional[int] = None, default: Optional[_T] = None) -> Union[Any, _T]:
```
Comment (addie9800) [inline src/fundus/parser/data.py:91]: Why does it make sense to update the return value to `Union[Any, _T]`?

**Review by addie9800 — APPROVED** [review]
Thanks for the update 👍

File: src/fundus/parser/data.py (hunk 1 above)
Comment (MaxDall) [reply src/fundus/parser/data.py:91]: So mypy does know that if a default is given, at least that type is returned.

Comment (addie9800) [reply src/fundus/parser/data.py:91]: Ah ok. I didn't know mypy was able to make that connection. Thanks

**Pull request merged**